packages/examples/cvat/exchange-oracle/tests/api/test_exchange_api.py

```diff
@@ -0,0 +1,283 @@
+                    "status",
+                }
+
+
+def test_list_tasks_401(client: TestClient) -> None:
+    response = client.get("/tasks", headers={"signature": "test"})
+
+    assert response.status_code == 401
+    assert response.json() == {"message": "Unauthorized"}
+
+    response = client.get(
+        "/tasks",
+    )
+
+    # Send a request without a signature
+    assert response.status_code == 400
+    assert response.json() == {"errors": [{"field": "signature", "message": "field required"}]}
+
+
+def test_register_200(client: TestClient) -> None:
+    with SessionLocal.begin() as session:
+        # New user
+        with patch("src.endpoints.exchange.cvat_api.get_user_id") as mock_get_user:
+            mock_get_user.return_value = 1
+            response = client.put(
+                "/register",
+                headers={"signature": "sample"},
+                json={"wallet_address": user_address, "cvat_email": cvat_email},
+            )
+
+        user = response.json()
+        assert response.status_code == 200
+        db_user = session.query(User).where(User.wallet_address == user_address).first()
+        assert user["wallet_address"] == db_user.wallet_address
+        assert user["cvat_id"] == db_user.cvat_id
+        assert user["cvat_email"] == db_user.cvat_email
+        assert user["wallet_address"] == user_address
+        assert user["cvat_email"] == cvat_email
+
+        # User with same address but different email
+        cvat_email_2 = "test2@hmt.ai"
+        with (
+            patch("src.endpoints.exchange.cvat_api.remove_user_from_org") as mock_remove_user,
+            patch("src.endpoints.exchange.cvat_api.get_user_id") as mock_get_user,
+        ):
+            mock_get_user.return_value = 1
+            response = client.put(
+                "/register",
+                headers={"signature": "sample"},
+                json={"wallet_address": user_address, "cvat_email": cvat_email_2},
+            )
+
+        user = response.json()
+        assert response.status_code == 200
+        mock_remove_user.assert_called_once()
+        assert user["wallet_address"] == user_address
+        assert user["cvat_email"] == cvat_email_2
+
+        # Existing user
+        cvat_email_2 = "test2@hmt.ai"
+        response = client.put(
+            "/register",
+            headers={"signature": "sample"},
+            json={"wallet_address": user_address, "cvat_email": cvat_email_2},
+        )
+
+        user = response.json()
+        assert response.status_code == 200
+        mock_remove_user.assert_called_once()
+        assert user["wallet_address"] == user_address
+        assert user["cvat_email"] == cvat_email_2
+
+
+def test_register_400(client: TestClient) -> None:
+    with SessionLocal.begin() as session:
+        # User with different wallet address
+        user = User(
+            wallet_address="0x86e83d346041E8806e352681f3F14549C0d2BC61",
+            cvat_email="test@hmt.ai",
```

Comment:
It's not clear from this code that these values are different from what's is in the `wallet_address` and `cvat_email` variables. Maybe it would be better to emphasize it somehow.